drivers/spi/spi_mcux_lpspi.c

```diff
@@ -739,29 +744,42 @@ static void spi_mcux_iodev_complete(const struct device *dev, int status)
 
 #endif
 
+#if defined(CONFIG_SPI_MCUX_LPSPI_DMA)
+static int lpspi_dma_dev_ready(const struct device *dma_dev)
+{
+	if (!device_is_ready(dma_dev)) {
+		LOG_ERR("%s device is not ready", dma_dev->name);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int lpspi_dma_devs_ready(struct spi_mcux_data *data)
+{
+	return lpspi_dma_dev_ready(data->dma_tx.dma_dev) |
```

Comment:
This is doing a binary OR on the return value of both functions. Right now this seems like it will work fine, because `-ENODEV | -ENODEV == -ENODEV`. However if the `lpspi_dma_dev_ready` function added multiple negative return codes for some reason, this would stop working. It seems like we should just return a bool from `lpspi_dma_dev_ready` and return `-ENODEV` in this function, rather than doing something tricky here.

Reply:
both of these functions are helper functions in this file, the function is defined 2 lines above this if that happens, the person making such a change will handle it, I'm not worried about this

Reply:
also, the signature of device_is_ready returns a bool, and im not sure what else that lpspi_dma_dev_ready would do besides check if the device is ready, so this seems like an extremely far fetched scenario